app/main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health=100, hidden=False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(animal):
+        if isinstance(animal, Herbivore) and not animal.hidden:
+            animal.health -= 50
+            if not animal.health:
```

Comment:
not very good conditions because i can create an animal with 30 xp, after biting it still will be in alive list in this case